commpare/cassandra/cassandra_utils.py

```diff
@@ -0,0 +1,265 @@
+        with temporary_cd(tmp_dir):
+            # Guess dihedral type...from contents of structure
+            if len(structure.dihedrals) > 0:
+                dihedral_style = 'charmm'
+            else:
+                dihedral_style = 'opls'
+            write_mcf(structure,mcf_file,angle_style='harmonic',
+                    dihedral_style=dihedral_style)
+            mb.load(structure).save(xyz_file)
+            mb.load(structure).to_trajectory().save_pdb(pdb_file)
+
+            inp_file,output = write_cassandra_inp()
+            successful_fraglib = run_fraglib_setup(fraglib_setup, cassandra,
+                                 inp_file, mcf_file, pdb_file, workdir)
+            if successful_fraglib:
+                run_cassandra(cassandra, inp_file, workdir)
+                energies = get_cassandra_energy(output + '.log')
+            else:
+                print("Cassandra failed due to unsuccessful fragment generation")
+
+    df = pd.DataFrame.from_dict(energies, orient='index')
+
+    return df
+
+def get_cassandra_energy(prpfile):
+    """ Parse and canonicalize energies from cassandra prp file
+
+    Notes
+    -----
+    Cassandra energy units are kJ/mol
+    Currently only support nonbond and all energy since Cassandra
+    does not support writing bond/angle/dihedral energies
+    """
+
+    cassandra_force_groups = {'cassandra': {}}
+    key_to_data = {'bond':['Bondenergy'],
+                  'angle':['Bondangleenergy'],
+               'dihedral':['Dihedralangleenergy'],
+                'nonbond':['Intramoleculevdw','Intramoleculeq','Intermoleculevdw',
+                           'Intermoleculeq', 'Reciprocalewald','Selfewald'],
+                    'all':['Totalsystemenergy']}
+
+    prp_dict = {}
+    read_lines = False
+    with open(prpfile) as f:
+        for line in f:
+            if line.strip() == 'Compute total energy':
+                read_lines = True
+            if read_lines == True and line.strip().split() != []:
+                key = ''
+                for word in line.strip().split()[:-1]:
+                    key += word
+                val = line.strip().split()[-1]
+                prp_dict[key] = val
+
+
+    #Factor to convert atomic energy (amu A^2/ ps^2) to kJ/mol
+    atomic_to_kJmol = 0.01
+    for canonical_name, cass_names in key_to_data.items():
+        if cass_names is not None:
+            cassandra_force_groups['cassandra'][canonical_name] = sum([
+                                    float(prp_dict[item])*atomic_to_kJmol
+                                    for item in cass_names
+                                    if item in prp_dict.keys()])
+        else:
+            cassandra_force_groups['cassandra'][canonical_name] = 'N/A'
+
+    return cassandra_force_groups
+
+def run_fraglib_setup(fraglib_setup,cassandra,inp_file,mcf_file,pdb_file,workdir):
+    fraglib_cmd = ('python2 {fraglib_setup} {cassandra} '.format(fraglib_setup=fraglib_setup,
```

Comment:
You need to pass your `py2` that you found from `detect_cassandra_binaries`

Reply:
Apologies for the dumb mistake. This should be updated now.